nano/node/socket.cpp

```diff
@@ -140,14 +149,20 @@ void nano::socket::checkup ()
 		{
 			uint64_t now (nano::seconds_since_epoch ());
 			auto condition_to_disconnect{ false };
-			if (this_l->is_realtime_connection () && (now - this_l->last_receive_time_or_init) > this_l->silent_connection_tolerance_time.count ())
+			auto is_silent = [this_l] (auto const & time) {
+				auto const silent_for_receiving = (time - this_l->last_receive_time_or_init) > this_l->silent_connection_tolerance_time.count ();
+				auto const silent_for_sending = (time - this_l->last_send_time_or_init) > this_l->silent_connection_tolerance_time.count ();
+				return silent_for_receiving && silent_for_sending;
```

Comment:
This shouldn't be an "AND" relationship.
It should at least be an OR relationship but I am not sure if we have a guaranteed minimum reply gap.